app/controllers/offices_controller.rb

```diff
@@ -1,22 +1,41 @@
 class OfficesController < ApplicationController
   def index
+    # return on direct requests to the page
     return unless search
-    return @error_message = search.error_message unless search.valid?
 
-    @transportation_offices = TransportationOffice.by_distance_with_shipping_office(search.result).paginate(page: params[:page])
+    # return error message on invalid search
+    return error_message unless search.valid?
+
+    # return search results on valid search (GET)
+    return transportation_offices unless request.post?
+
+    # redirect on valid search (POST)
+    redirect_to offices_path(search.result)
   end
 
   private
 
+  def coordinates_search
+    CoordinatesSearch.new(params) if params[:latitude].present? && params[:longitude].present?
+  end
+
+  def error_message
+    @error_message ||= search.error_message
+  end
+
+  def installation_search
+    InstallationSearch.new(params) if params[:query].present?
+  end
+
   def search
-    @search ||= coordinates_search || zip_code_search || nil
+    @search ||= coordinates_search || zip_code_search || installation_search || nil
   end
 
-  def coordinates_search
-    CoordinatesSearch.new(params) if params[:coordinates].present?
+  def transportation_offices
+    @transportation_offices ||= TransportationOffice.by_distance_with_shipping_office(search.result).paginate(page: params[:page])
```

Comment:
curious why you're storing this in a var if it's only called once
also, could we change by_distance_with_shipping_office to add a result size limit?

Reply:
This method exposes `@transportation_offices` to the view (see lines 37, 64, 67, and 70).

> could we change by_distance_with_shipping_office to add a result size limit?

We could certainly limit the results set. There's absolutely limited utility beyond `x` results. Do you have any thoughts on a usable limit?

Reply:
hmm how about 20

Reply:
Just thought of something. We'll soon be adding weight scales to the map and list of results and interleaving them with office locations. That'll have some impact on how we show results.

Mind if I punt on limiting results for now and we can address it in the near-ish future?

Reply:
sure, no rush for that